### PR TITLE
MGMT-9451: Moving to new service-agent protocol

### PR DESCRIPTION
**Pull request by tsorya**
[MGMT-9451](https://issues.redhat.com/browse/MGMT-9451): Moving to new service-agent protocol
Moving new commands : stop, logs, install and next_step

Continuing moving commands from service to agent

### CODE REVIEW

File: src/commands/actions/install_cmd.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	if a.installParams.InstallerArgs != "" {
+		installerCmdArgs = append(installerCmdArgs, "--installer-args", a.installParams.InstallerArgs)
+	}
+
+	noProxyArgs := getProxyArguments(a.installParams.Proxy)
+	if len(noProxyArgs) > 0 {
+		installerCmdArgs = append(installerCmdArgs, noProxyArgs...)
+	}
```
Comment (eranco74) [inline src/commands/actions/install_cmd.go:98]:
```suggestion
	proxyArgs := getProxyArguments(a.installParams.Proxy)
	if len(proxyArgs) > 0 {
		installerCmdArgs = append(installerCmdArgs, proxyArgs...)
	}
```

File: src/commands/actions/install_cmd.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	if len(a.installParams.ServiceIps) > 0 {
+		installerCmdArgs = append(installerCmdArgs, "--service-ips", strings.Join(a.installParams.ServiceIps, ","))
+	}
+
+	return fmt.Sprintf("%s %s %s", shellescape.QuoteCommand(podmanCmd), swag.StringValue(a.installParams.InstallerImage),
+		shellescape.QuoteCommand(installerCmdArgs))
+}
+
+func getProxyArguments(proxy *models.Proxy) []string {
+	cmd := make([]string, 0)
```
Comment (eranco74) [inline src/commands/actions/install_cmd.go:109]: Change `cmd` to `proxyArgs` in this function

File: src/commands/actions/install_cmd.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+type install struct {
+	args          []string
+	installParams models.InstallCmdRequest
+}
+
+func (a *install) Validate() error {
+	err := validateCommon("install", 1, a.args, &a.installParams)
+	if err != nil {
+		return err
+	}
```
Comment (eranco74) [inline src/commands/actions/install_cmd.go:32]: Create a ticket for improving the validation of this command

**Review by eranco74 — COMMENTED** [review]
looks good.
It's a bit strange that the `Run` functions don't actually run anything.
They just return the command and args, consider renaming...

File: src/commands/actions/install_cmd.go (hunk 3 above)
Comment (tsorya) [reply src/commands/actions/install_cmd.go:32]: https://issues.redhat.com/browse/MGMT-9453

**Comment by tsorya** [comment]
> Member

changed to CreateCmd

**Review by eranco74 — COMMENTED** [review]
image

/lgtm

**Pull request merged**